modules/social_features/social_profile/modules/social_profile_privacy/src/Service/SocialProfilePrivacyBatchHelper.php

```diff
@@ -0,0 +1,236 @@
+<?php
+
+namespace Drupal\social_profile_privacy\Service;
+
+use Drupal\Core\Batch\BatchBuilder;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Extension\ModuleHandlerInterface;
+use Drupal\profile\Entity\ProfileInterface;
+
+/**
+ * Class SocialProfilePrivacyBatchHelper.
+ *
+ * Update profile names in batch.
+ *
+ * @package Drupal\social_profile_privacy
+ */
+class SocialProfilePrivacyBatchHelper {
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * The module handler.
+   *
+   * @var \Drupal\Core\Extension\ModuleHandlerInterface
+   */
+  protected $moduleHandler;
+
+  /**
+   * SocialProfilePrivacyBatchHelper constructor.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager.
+   * @param \Drupal\Core\Extension\ModuleHandlerInterface $module_handler
+   *   The module handler.
+   */
+  public function __construct(
+    EntityTypeManagerInterface $entity_type_manager,
+    ModuleHandlerInterface $module_handler
+  ) {
+    $this->entityTypeManager = $entity_type_manager;
+    $this->moduleHandler = $module_handler;
+  }
+
+  /**
+   * Get profile name.
+   *
+   * @param \Drupal\profile\Entity\ProfileInterface|null $profile
+   *   The profile.
+   *
+   * @return string|void
+   *   The generated profile name value.
+   *
+   * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
+   * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
+   */
+  public function getProfileName(ProfileInterface $profile = NULL) {
+    /** @var \Drupal\user\UserStorageInterface $user_storage */
+    $user_storage = $this->entityTypeManager->getStorage('user');
+
+    // Do nothing if no profile.
+    if ($profile == NULL) {
+      return '';
+    }
+
+    /** @var \Drupal\user\UserInterface $account */
+    $account = $user_storage->load($profile->getOwnerId());
+
+    // Set default profile name.
+    $account_name = $account->getAccountName();
+
+    // Get profile private fields list.
+    $uid = $account->id();
+    $private_fields_list = social_profile_privacy_private_fields_list($uid);
+
+    $account_name_fields = [
+      'field_profile_first_name',
+      'field_profile_last_name',
+    ];
+
+    foreach ($account_name_fields as $key => $account_name_field) {
+      if (in_array($account_name_field, $private_fields_list)) {
+        unset($account_name_fields[$key]);
+      }
+      else {
+        $account_name_fields[$key] = $profile->get($account_name_field)->getString();
```

Comment:
It would be nicer if use a different variable name for storing the values. :-)

Reply:
Updated.